tests/core/newton_non_linear_solver_01.cc

```diff
@@ -1,29 +1,13 @@
-#include <deal.II/lac/full_matrix.h>
-#include <deal.II/lac/lapack_full_matrix.h>
-#include <deal.II/lac/sparse_direct.h>
-#include <deal.II/lac/vector.h>
-
-#include <core/newton_non_linear_solver.h>
-#include <core/physics_solver.h>
-
-#include <iostream>
-#include <memory>
-
-#include "../tests.h"
```

Comment:
Are these includes really non-necessary? Then where are they drawn from? I am always of cross-include dependency. Clearly they are not necessary in the .cc, but you are sure they are not needed in the .h?